src/actions/app.js

```diff
@@ -24,10 +22,10 @@ export const navigate = (path, params) => (dispatch) => {
 };
 
 const loadPage = (page, subpage, tab) => (dispatch) => {
-    const pageInfo = pageInfoCollection[page];
+    const importInfo = pageImportInfoCollection[page];
 
-    if (pageInfo && (!pageInfo.requiresSubpage || pageInfo.subpages[subpage])) {
-        pageInfo.importTarget();
+    if (importInfo && (!importInfo.requiresSubpage || importInfo.subpages[subpage] && !importInfo.subpages[subpage].hasMoved)) {
```

Comment:
quite complicated condition, can we extract some part into a function with a descriptive name?